tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -218,6 +218,51 @@ TILEDB_DEPRECATED_EXPORT int32_t tiledb_query_add_point_ranges(
     const void* start,
     uint64_t count);
 
+/* ********************************* */
+/*        QUERY STATUS DETAILS       */
+/* ********************************* */
+
+/** This should move to c_api/tiledb.h when stabilized */
+typedef struct tiledb_query_status_details_t tiledb_query_status_details_t;
+
+/** TileDB query status details type. */
+typedef enum {
+/** Helper macro for defining status details type enums. */
+#define TILEDB_QUERY_STATUS_DETAILS_ENUM(id) TILEDB_##id
+#include "tiledb_enum.h"
+#undef TILEDB_QUERY_STATUS_DETAILS_ENUM
+} tiledb_query_status_details_reason_t;
+
+/** This should move to c_api/tiledb_struct_defs.h when stabilized */
+struct tiledb_query_status_details_t {
+  tiledb_query_status_details_reason_t reason;
```

Comment:
Possibly rename this to incomplete_reason for clarity?

Reply:
Updated